packages/web3-eth-abi/src/api/parameters_api.ts

```diff
@@ -41,8 +41,10 @@ import { encodeParameters as encodeParametersInternal } from '../coders/encode.j
  *  > 0x000000000000000000000000000000000000000000000000000000008bd02b7b0000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000748656c6c6f212500000000000000000000000000000000000000000000000000
  * ```
  */
-export const encodeParameters = (abi: ReadonlyArray<AbiInput>, params: unknown[]): string =>
-	encodeParametersInternal(abi, params);
+export const encodeParameters = (
+	abi: ReadonlyArray<AbiInput> | 'infer-types',
```

Comment:
is there a particular reason we are passing `infer-types` for the first paramter instead of having it as a new optional parameter?

Reply:
No specific reason except that the caller still need to pass something for this first parameter. And using for example `undefined` used to cause to throw and I did not want to change the behavior. 
However, thanks for commenting on this ugly param. I now created a separate function to handle the infer-types logic.